### PR TITLE
Add exporter for tax categories

### PR DESCRIPTION
**Pull request by solutionDrive-Alt**
Missing Tests for now, started as quick test for #160. It works as expected.
Edit: Behat-Tests are now present

### CODE REVIEW

**Comment by solutionDrive-Alt** [comment]
fixes also #162. I also realized that our behat-tests for cli-export do not test anything useful. They just ensure that the command can be executed, but the resulting files are empty (only the `header`-line is written) because we don't have any data in the db at this moment. 

I solved that for `tax_category` in this PR an will apply those changes to the other resources with the next PR.

File: composer.json [hunk 1]
```diff
@@ -58,6 +59,9 @@
     "config": {
         "bin-dir": "bin"
     },
+    "conflict": {
+        "sonata-project/core-bundle": ">=3.12"
```
Comment (stefandoorn) [inline composer.json]: Are all higher versions conflicting or just a single release?

File: tests/Behat/Context/CliBaseContext.php [hunk 2]
```diff
@@ -127,4 +142,21 @@ public function iExportDataToSpecificFiletypeFileWithTheCliCommand($exporterType
...
+        $actualContent = file($this->exportFile);
+        foreach ($actualContent as $index => $actualRowInFile) {
+            // Check if the line in the exported file differs from the expected one
+            Assert::assertTrue($this->getDiff($expectedContent, $index, $actualRowInFile) === 0);
+        }
+    }
+
+    private function getDiff(TableNode $table, int $index, string $rowInFile): int
+    {
+        return count(array_diff($table->getRow($index), explode(';', trim($rowInFile))));
```
Comment (stefandoorn) [inline tests/Behat/Context/CliBaseContext.php]: Maybe better use [str_getcsv](http://php.net/manual/en/function.str-getcsv.php)?

File: tests/Behat/Context/CliTaxCategoriesContext.php [hunk 3]
```diff
@@ -19,4 +20,20 @@ public function iShouldHaveAtLeastTheFollowingTaxCategoriesIdsInTheDatabase(Tabl
             Assert::assertNotNull($taxCategory);
         }
     }
+
+    /**
+     * @Then /^(this tax category) name is "([^"]+)"$/
+     */
+    public function thisTaxCategoryNameShouldBe(TaxCategoryInterface $taxCategory, $taxCategoryName)
```
Comment (stefandoorn) [inline tests/Behat/Context/CliTaxCategoriesContext.php]: Missing void return

File: tests/Behat/Context/CliTaxCategoriesContext.php [hunk 4]
```diff
@@ -19,4 +20,20 @@ public function iShouldHaveAtLeastTheFollowingTaxCategoriesIdsInTheDatabase(Tabl
...
+    {
+        $taxCategory->setName($taxCategoryName);
+    }
+
+    /**
+     * @Then /^(this tax category) description is "([^"]+)"$/
+     */
+    public function thisTaxCategoryDescriptionShouldBe(TaxCategoryInterface $taxCategory, $taxCategoryName)
```
Comment (stefandoorn) [inline tests/Behat/Context/CliTaxCategoriesContext.php]: Missing void return

File: composer.json (hunk 1 above)
Comment (solutionDrive-Alt) [reply composer.json]: 3.13.4 is working fine. Should I test the versions in between?

File: tests/Behat/Context/CliBaseContext.php (hunk 2 above)
Comment (solutionDrive-Alt) [reply tests/Behat/Context/CliBaseContext.php]: Just saw that you wrote `str_getcsv`... :-) But I think `fgetcsv` is also an improvement. :-)

File: tests/Behat/Context/CliBaseContext.php [hunk 5]
```diff
@@ -69,12 +71,23 @@ class CliBaseContext implements Context
...
+     */
+    protected $objectManager;
+
     public function __construct(
         KernelInterface $kernel,
         RepositoryInterface $repository,
         ImporterRegistry $importerRegistry,
         ExporterRegistry $exporterRegistry,
         ContainerInterface $container,
+        ObjectManager $objectManager,
```
Comment (lsmith77) [inline tests/Behat/Context/CliBaseContext.php]: why was this added?

File: tests/Application/yarn.lock [hunk 6]
```diff
@@ -24,6 +24,15 @@ ajv@^4.9.1:
     co "^4.6.0"
     json-stable-stringify "^1.0.1"
 
+ajv@^6.5.5:
```
Comment (lsmith77) [inline tests/Application/yarn.lock]: why was this updated?

File: tests/Behat/Context/CountriesContext.php [hunk 7]
```diff
@@ -6,9 +6,9 @@
 
 use Behat\Behat\Context\Context;
 use Behat\Mink\Session;
+use FriendsOfBehat\PageObjectExtension\Page\SymfonyPage;
 use PHPUnit\Framework\Assert;
 use Sylius\Behat\Context\Transform\CountryContext;
-use Sylius\Behat\Page\SymfonyPage;
```
Comment (lsmith77) [inline tests/Behat/Context/CountriesContext.php]: should this change be done in other places too (in another PR)?

Comment (solutionDrive-Alt) [reply tests/Behat/Context/CountriesContext.php]: Yeah, at that time I planned to fixed this for the other Pages in a follow-up PR.

File: tests/Behat/Context/CliBaseContext.php (hunk 5 above)
Comment (solutionDrive-Alt) [reply tests/Behat/Context/CliBaseContext.php]: I think I forgot to remove this, when I was cleaning up this PR. I also realized at this time, that our behat tests for the export aren't very useful at the moment, because they don't test more than command-execution. #163

File: tests/Application/yarn.lock (hunk 6 above)
Comment (solutionDrive-Alt) [reply tests/Application/yarn.lock]: Honestly, I don't recall updating this. Maybe also related to #163 . I will check this later today or this weekend.

Comment (solutionDrive-Alt) [reply tests/Application/yarn.lock]: it looks like it is related to the update of sylius from version `1.2.8` to `1.2.12`, which I have done. Shall I revert this (update of sylius)?

File: tests/Behat/Context/CliBaseContext.php (hunk 5 above)
Comment (solutionDrive-Alt) [reply tests/Behat/Context/CliBaseContext.php]: I removed this with and all injections into the services.

File: tests/Behat/Context/CountriesContext.php (hunk 7 above)
Comment (solutionDrive-Alt) [reply tests/Behat/Context/CountriesContext.php]: actually this is the only Context which extends SymfonyPage, and I'm not sure if this is necessary at all, since it is a `Context` and not a `PageObject`. Every other context is using `ResourceIndexPageInterface` which extends the `\Sylius\Behat\Page\Admin\Crud\IndexPage`. And this already uses the correct PageObject of the `FriendsOfBehat\PageObjectExtension`. Problably this was also the reason for the `Sylius`-update I made.

File: tests/Behat/Context/CliBaseContext.php (hunk 2 above)
Comment (solutionDrive-Alt) [reply tests/Behat/Context/CliBaseContext.php]: @stefandoorn is this also ok, for you or is there a reason why I should change it to `str_getcsv`?

Comment (stefandoorn) [reply tests/Behat/Context/CliBaseContext.php]: Yeah if it works :-) I try to use native functions for these things instead of rolling it myself, but I think this does the job :-)

File: tests/Behat/Context/CountriesContext.php (hunk 7 above)
Comment (stefandoorn) [reply tests/Behat/Context/CountriesContext.php]: Fixed in #168

**Comment by lsmith77** [comment]
please rebase on master

**Comment by solutionDrive-Alt** [comment]
On it... :-)

**Comment by solutionDrive-Alt** [comment]
@stefandoorn I think I addressed all feedback.

**Comment by solutionDrive-Alt** [comment]
@lsmith77 @stefandoorn I will merge this since both of you approved.
